recordset/recordset.controller.js

```diff
@@ -114,40 +111,68 @@
             return elements;
         }
 
+        function setRecordsetHeight() {
+            var elements = fetchMainElements();
+            // if these 2 values are not set yet, don't set the height
+            if (elements.navbarHeight && elements.bookmarkHeight) {
+                UiUtils.setDisplayContainerHeight(elements);
+                // no need to fetch and verify the faceting elements (navbar and bookmark are the same container as the ones used in main elements function)
+                if (chaiseConfig.showFaceting) UiUtils.setDisplayContainerHeight(fetchFacetingElements());
+            }
+        }
+
+        function fetchBodyElements() {
```

Comment:
In all three apps we have `fetchMainElements` and `fetchBodyElements`. I think you added `fetchBodyElements` because you were computing the height of `main-body` by listing all of its children. But now it seems unnecessary and we can add these attributes to the other function.